src/pydsl/compiler.py

```diff
@@ -239,7 +239,6 @@ class ToMLIR(ToMLIRBase):
     # MLIR programs being generated multiple times
     # ast.NodeVisitor cannot modify the tree it is visiting, so cache will
     # never be outdated
-    @cache
```

Comment:
As discussed, we are going to re-introduce this since it breaks @tag, we will mark another test case as xfail.